mock/Cargo.toml

```diff
@@ -0,0 +1,23 @@
+[package]
+name = "miden-mock"
+version = "0.1.0"
+description = "Core components of the Miden rollup"
+authors = ["miden contributors"]
+readme = "README.md"
+license = "MIT"
+repository = "https://github.com/0xPolygonMiden/miden-base"
+categories = []
+keywords = []
+edition = "2021"
+rust-version = "1.67"
+
+[dependencies]
+clap = { version = "4.4" }
+getrandom = { version = "0.2", features = ["js"] }
+rand = { version = "0.8" }
+rand_pcg = { version = "0.3", features = ["serde1"] }
+miden-objects = { path = "../objects" , features = ["mock", "serde", "log"]}
```

Comment:
To produce the mock chain this binary needs the `mock` feature. This will in turn enable the `testing` feature, which will eventually enable winter utils. Winter utils is not wasm compatible, which caused the wasm CI build to fail.

To fix the above, I added `internal` modules for the mocks that are not wasm32 compatible.

Reply:
Do you mean `winter-rand-utils`? (regular `winter-utils` should be WASM-compatible).

Reply:
Yes, I mean `winter-rand-utils`, [ref](https://github.com/facebook/winterfell/blob/main/utils/rand/src/lib.rs#L13-L14)

Reply:
What's the relationship between `mock` and `testing` feature? Without looking too deeply into this, it seems like `testing` should require `mock` but not the other way around.

This probably doesn't really change things, as we won't be able to generate mock data using WASM anyway. The one thing that always confuses me is that I always think that we shouldn't be able to do WASM target for `std` compilation, and thus, we just need the `std` flag.

Reply:
> What's the relationship between mock and testing feature? Without looking too deeply into this, it seems like testing should require mock but not the other way around.

Don't remember why it was enabled, I guess it had to do with the assembler utility used to generate account ids being available under `testing` flag, or with the use of the rand utilities from winterfell that was requested somewhere.

It seems it is no longer necessary, so I updated the `Cargo.toml` on the `objects` crate, and the `testing` and `mock` flags don't depend on each other anymore.

Reply:
I had to re-add testing as a dependency to mock. It seems the original `mock` mod was originally created behind the `testing` flag and there some dependencies that need fixing (the assembler as mentioned above is one):

https://github.com/0xPolygonMiden/miden-base/commit/3440924f2a0e82bc0109460742550451dec2d9ab#diff-91dfa434fdd244ed04288d28f4748f567ae1e75d653dc1fcc0cf6fe6f41a8ed1R51-R52